python/cudf/cudf/core/dataframe.py

```diff
@@ -394,6 +394,7 @@ def mask(self, other):
         for col in self.columns:
             if col in other.columns:
                 boolbits = cudautils.compact_mask_bytes(
+                    # leave this alone
                     other[col].to_gpu_array()
```

Comment:
Why does this need to be left alone? I figure other here is a Series of booleans which we shouldn't need to return an array of by copy.

Reply:
We should either assert there's no nulls or gracefully handle nulls via a `fillna`